apps/portal/components/copy-button.tsx

```diff
@@ -0,0 +1,40 @@
+import { Copy } from "lucide-react";
+
+import { useToast } from "@recallnet/ui/hooks/use-toast";
+import { cn } from "@recallnet/ui/lib/utils";
+
+interface CopyButtonProps {
+  value: string;
+  tooltip?: string;
+  successMessage?: string;
+  className?: string;
+  iconClassName?: string;
+}
+
+export function CopyButton({
+  value,
+  tooltip = "Copy to clipboard",
+  successMessage = "Copied to clipboard",
+  className,
+  iconClassName,
+}: CopyButtonProps) {
+  const { toast } = useToast();
+
+  const handleCopy = (e: React.MouseEvent<HTMLDivElement>) => {
+    e.preventDefault();
+    navigator.clipboard.writeText(value);
+    toast({ title: successMessage ?? "Copied to clipboard" });
+  };
+
+  return (
+    <div
+      title={tooltip}
+      onClick={handleCopy}
+      className={cn("cursor-pointer", className)}
+    >
+      <Copy
+        className={cn("size-4 opacity-20 hover:opacity-100", iconClassName)}
+      />
+    </div>
```

Comment:
The props and view hierarchy here could be simplified. The outer div can really do nothing except for provide the `title` prop so the tooltip displays. The real content of this component is the `Copy` component. You should make the props for the whole component extend the prop type of `Copy` and allow all `className` modifications to happen just on the `Copy` element. The props would look like:

```
type CopyButtonProps = {
  value: string;
  tooltip?: string;
  successMessage?: string;
} & HTMLAttributes<SVGElement>;
```
Then users can style the svg any way they want.

Reply:
okay, i think i got all of 'em